File: prove/docker/pypy/server/MicroServer.py
import grpc
import time

from concurrent import futures

from proto import ClientToServer_pb2
from proto import ClientToServer_pb2_grpc

class GreeterServicer(ClientToServer_pb2_grpc.GreeterServicer):

    def GetServiceName(self, Request, context):
        print("Request: ", Request.name)
        output = ClientToServer_pb2.Response(serviceName = "M2")
        return output

#create gRPC server
server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
ClientToServer_pb2_grpc.add_GreeterServicer_to_server(GreeterServicer(), server)

print('Starting server. Listening on port 50051.')
server.add_insecure_port('[::]:50051')
server.start()

try:
    while True:
        time.sleep(86400)   #86400 seconds == 24 hours
except KeyboardInterrupt:
    server.stop(0)

File: prove/docker/gogo/server/MicroServer.go
package main

import (
	"context"
	"log"
	"net"
	"fmt"

	"google.golang.org/grpc"
	pb "server/gen"
)

type GreeterServer struct {
	pb.UnimplementedGreeterServer
}

func (s *GreeterServer) GetServiceName(ctx context.Context, req *pb.Request) (*pb.Response, error) {
	log.Println("Request: ", req.Name)
	return &pb.Response{ServiceName: "M2"}, nil
}

func main() {
	listener, err := net.Listen("tcp", "localhost:50051")
	if err != nil {
		log.Fatalln(err)
	}
	fmt.Println("Starting server. Listening on port 50051.")

	//create gRPC server
	grpcServer := grpc.NewServer()

	pb.RegisterGreeterServer(grpcServer, &GreeterServer{})

	grpcServer.Serve(listener)
	if err != nil {
		log.Println(err)
	}

}
